tests/functional/ec2-rds-dbinstance-with-vpc-sg.nix

```diff
@@ -0,0 +1,30 @@
+let
+  region = "us-east-1";
+in
+{
+  network.description = "NixOps Test";
+
+  resources.rdsDbInstances.test-rds-instance =
+    { resources, ... }:
+    {
+      inherit region;
+      id = "myOtherDatabaseIsAFilesystem";
+      instanceClass = "db.r3.large";
+      allocatedStorage = 30;
+      masterUsername = "administrator";
+      masterPassword = "testing123";
+      port = 5432;
+      engine = "postgres";
+      dbName = "helloDarling";
+      vpcSecurityGroups = [ resources.ec2SecurityGroups.test-rds-sg ];
+    };
+
+  resources.ec2SecurityGroups.test-rds-sg =
```

Comment:
I guess this should work on new AWS accounts that use default vpc but for EC2 classic accounts the sg won't be a vpc one ?
I think you could add a vpc as resource here and pass it to vpcId attribute of the security group ?

Reply:
@AmineChikhaoui this turns out to be trickier than that. Running from EC2-VPC account if I introduce a new VPC and assign a security group to that VPC then I also have to ensure that an RDS instance is created in the same VPC. To be able to do that we need to introduce DB Subnet Group resource (which currently is not available with NixOps) and extend RDS resource to support DB Subnet Group assignment. In a real production setup we need this functionality anyway, otherwise we will be locked to a default VPC. 

With that said, do you think this work needs to be done in this or separate PR?

Reply:
@alexeiskachykhin I think for this PR you could start by adding an option for dbSubnetGroup that takes a string first so that we can create an rds dbinstance in a VPC != default. Later a separate resource can be implemented.